cpp/src/arrow/sparse_tensor.h

```diff
@@ -111,22 +111,46 @@ class ARROW_EXPORT SparseCOOIndex : public internal::SparseIndexBase<SparseCOOIn
  public:
   static constexpr SparseTensorFormat::type format_id = SparseTensorFormat::COO;
 
-  /// \brief Make SparseCOOIndex from raw properties
+  /// \brief Make SparseCOOIndex from a coords tensor and canonicality
```

Comment:
It might be useful to add a definition of what "canonicality" is at the top of the class. This can be done later